src/resmom/linux/alps.c

```diff
@@ -4530,8 +5274,19 @@ alps_create_reserve_request(job *pjob, basil_request_reserve_t **req)
 
 		p->depth = ns->depth;
 		p->nppn = width = ns->width;
```

Comment:
PBS uses only one BASIL version when interfacing with ALPS.  Thus if PBS can read-in and parse BASIL 1.3 it will make ALPS reservations using BASIL 1.3.  But if PBS can only read-in and understand BASIL 1.2, then PBS will make ALPS reservations using BASIL 1.2.  And if PBS uses BASIL 1.2 then there would be no need for nppcu to be sent, and ALPS will provide the behavior that PBS expects.
The reason this may have been working for you is that you had also reduced the node's processor count and it matches up with what ALPS will allow for reservations.  Without nppcu set (or set to 0) ALPS will allow you to use all of the physical processors, which matched up with you counting CUs.
I am confident that once you fix how you count your processors, you will then need to specify nppcu.  You can test my theory by counting the processors as I have indicated, but don't set nppcu.  Then submit a job that requests all of the ncpus that PBS reports on a vnode (choose a vnode that has more than one processor per compute unit) and see what happens.  I think you're going to get the ALPS error "in user NIDs request exceeds max resources"